core/server/master/src/main/java/alluxio/master/file/activesync/ActiveSyncer.java

```diff
@@ -80,32 +82,45 @@ public void heartbeat() {
           // This returns a list of ufsUris that we need to sync.
           Set<AlluxioURI> ufsSyncPoints = syncInfo.getSyncPoints();
           for (AlluxioURI ufsUri : ufsSyncPoints) {
-            AlluxioURI alluxioUri = mMountTable.reverseResolve(ufsUri).getUri();
-            if (alluxioUri != null) {
-              if (syncInfo.isForceSync()) {
-                LOG.debug("force full sync {}", ufsUri.toString());
-                RetryUtils.retry("Full Sync", () -> {
-                  mFileSystemMaster.activeSyncMetadata(alluxioUri, null,
-                      mSyncManager.getExecutor());
-                }, RetryUtils.defaultActiveSyncClientRetry(ServerConfiguration
-                    .getMs(PropertyKey.MASTER_UFS_ACTIVE_SYNC_POLL_TIMEOUT)));
-              } else {
-                LOG.debug("sync {}", ufsUri.toString());
-                RetryUtils.retry("Incremental Sync", () -> {
-                  mFileSystemMaster.activeSyncMetadata(alluxioUri,
-                      syncInfo.getChangedFiles(ufsUri).stream().parallel()
-                          .map((uri) -> mMountTable.reverseResolve(uri).getUri())
-                          .collect(Collectors.toSet()),
-                      mSyncManager.getExecutor()
-                  );
-                }, RetryUtils.defaultActiveSyncClientRetry(ServerConfiguration
-                    .getMs(PropertyKey.MASTER_UFS_ACTIVE_SYNC_POLL_TIMEOUT)));
+            // Parallelize across sync points
+            List<Callable<Void>> tasksPerSyncPoint = new ArrayList<>(ufsSyncPoints.size());
+            tasksPerSyncPoint.add(() -> {
```

Comment:
There is a lot of testing here. Can we split this into a different method, so it is easier to understand the code?

Reply:
Thanks! Much easier to follow!